internal/driver/internal/k8s/logicalvolume_service.go

```diff
@@ -301,48 +304,60 @@ func (s *LogicalVolumeService) CreateSnapshot(ctx context.Context, node, dc, sou
 
 // ExpandVolume expands volume
 func (s *LogicalVolumeService) ExpandVolume(ctx context.Context, volumeID string, requestBytes int64) error {
-	logger.Info("k8s.ExpandVolume called", "volumeID", volumeID, "size", requestBytes)
+	logger := logger.WithValues("volume_id", volumeID, "size", requestBytes)
+	logger.Info("k8s.ExpandVolume called")
+	request := resource.NewQuantity(requestBytes, resource.BinarySI)
 
 	lv, err := s.GetVolume(ctx, volumeID)
 	if err != nil {
 		return err
 	}
 
-	err = s.updateSpecSize(ctx, volumeID, resource.NewQuantity(requestBytes, resource.BinarySI))
+	err = s.updateSpecSize(ctx, volumeID, request)
 	if err != nil {
 		return err
 	}
 
-	// wait until topolvm-node expands the target volume
-	for {
-		logger.Info("waiting for update of 'status.currentSize'", "name", lv.Name)
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		case <-time.After(1 * time.Second):
-		}
-
+	return wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
+		Duration: 1 * time.Second, // initial backoff
+		Factor:   2,               // factor for duration increase
+		Jitter:   0.1,
+		Steps:    math.MaxInt, // run for infinity; we assume context gets canceled
+	}, func(ctx context.Context) (bool, error) {
 		var changedLV topolvmv1.LogicalVolume
-		err := s.getter.Get(ctx, client.ObjectKey{Name: lv.Name}, &changedLV)
-		if err != nil {
+		if err := s.getter.Get(ctx, client.ObjectKey{Name: lv.Name}, &changedLV); err != nil {
 			logger.Error(err, "failed to get LogicalVolume", "name", lv.Name)
-			return err
+			return false, err
 		}
+
+		if changedLV.Spec.Size.Cmp(*request) != 0 {
+			logger.Info("waiting for update of 'spec.size' to propagate "+
+				"to signal requested expansion", "name", lv.Name)
+			return false, nil
+		}
```

Comment:
To be clear, why is this needed? We updated the spec [here](https://github.com/topolvm/topolvm/pull/903/files#diff-f3c95da9af2be35fd4a88902d32407e715af7b28b89c590c97f9d3e5f8d96453R316), I assume this is redundant.

Reply:
Usually this is avoided in controllers because it is recommended to just return and wait for the watcher to update after updating a resource in the reconcile loop, but we dont have that.

Reply:
Get it. BTW, we already know the request size, so why don't you use the `request` variable directly [here](https://github.com/topolvm/topolvm/pull/903/files/51d0ca59d9a2fcf59eaefbb63b6882181047e839#diff-f3c95da9af2be35fd4a88902d32407e715af7b28b89c590c97f9d3e5f8d96453L339)?

Reply:
I think ultimately it does not matter if I use request because above I checked that spec.Size == request. If its important to you I can adjust it.

Reply:
I mean lines from 334 to 337 and lines from 351 to 356 can be merged. We know `changedLV.Status.CurrentSize` will eventually equal `request`, could we write the code as such?

Reply:
Got what you mean now, changed accordingly, youre completely right.